Meadowlark-js/packages/meadowlark-core/src/security/JwtStatus.ts

```diff
@@ -15,6 +15,8 @@ export interface JwtStatus {
   subject: string | null;
   issuedAt: number;
   expiresAt: number;
+  roles: string[];
+  authorizationStrategy: string;
```

Comment:
Why type this as a `string` instead of `AuthorizationStrategy`?

Reply:
As is, AuthoruzationStrategy doesn't have an UNKNOWN (or similar) type of strategy, so nothing to set it to if it can't be parsed. I wasn't sure if adding to AuthoruzationStrategy was the correct way to go, but if you don't see a problem with it, it would probably be cleaner

Reply:
If it can't be parsed to a known strategy, then isn't that an error that should generate a 500 response?

Reply:
I was missing a couple of things in the code. We do have a UNDEFINED AuthorizationStrategy, so I was able to update this to not use a string (and not need the empty string above). But your question makes me wonder - what is the apropriate error response to throw if there is role defined and thus the token is bad. I think a 500 would be incorrect here and it should be a 400, 401 or 403 (currently it's a 401)

Reply:
Here's why I thought of 500: right now, that means that our code that _generated_ the token did something wrong. But since this is at a latter point, and that code to generate the token could be 3rd party in the future, I agree with you that 500 is actually not appropriate. Its simply a bad token and therefore unauthorized. Ideally, our code would include a message stating that the token is missing role information. Would that be easy to do in this ticket? Or better to put into a stand-alone ticket?

Reply:
![image](https://user-images.githubusercontent.com/48261255/180840797-48a85cbc-3e33-4d9c-a071-8329d2b0580c.png)
This is the output when roles can not be parsed

Reply:
Interestingly, because it would be rare to have an invalid token and yet have roles, this error is now basically the default error if the token is bad